### PR TITLE
New package: py-cx-oracle

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cx-oracle/package.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+class PyCxOracle(PythonPackage):
+    """Python interface to Oracle"""
+
+    homepage = "https://oracle.github.io/python-cx_Oracle"
+    pypi     = "cx_Oracle/cx_Oracle-8.3.0.tar.gz"
+
+    version('8.3.0', sha256='3b2d215af4441463c97ea469b9cc307460739f89fdfa8ea222ea3518f1a424d9')
+
+    depends_on('python@3.5:', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cx-oracle/package.py]:
```suggestion
    depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-cx-oracle/package.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    depends_on('oracle-instant-client', type=('build', 'link', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cx-oracle/package.py]: Where do you see this dep listed?

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cx-oracle/package.py (hunk 2 above)
Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cx-oracle/package.py]: It is needed at runtime at least:

> Add Oracle 21, 19, 18, 12 or 11.2 client libraries to your operating system library search path such as PATH on Windows or LD_LIBRARY_PATH on Linux

(from [here](https://cx-oracle.readthedocs.io/en/latest/user_guide/installation.html#quick-start-cx-oracle-installation))

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cx-oracle/package.py]: But looks like it's not needed at build-time

**Comment by iarspider** [comment]
Need to test exact dependency type (by connecting to a real Oracle db)

**Review by adamjstewart — APPROVED** [review]

**Comment by adamjstewart** [comment]
Want me to wait before merging?

**Comment by iarspider** [comment]
@adamjstewart yes, I'm testing it now.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**